include/onnxruntime/core/session/onnxruntime_cxx_api.h

```diff
@@ -992,6 +1012,63 @@ struct SessionOptions : detail::SessionOptionsImpl<OrtSessionOptions> {
 };
 
+#if !defined(ORT_MINIMAL_BUILD)
+/** \brief Options object used when compiling a model.
+ *
+ * Wraps ::OrtModelCompilationOptions object and methods
+ */
+struct ModelCompilationOptions;
+
+namespace detail {
+
+template <typename T>
+struct ConstModelCompilationOptionsImpl : Base<T> {
+  using B = Base<T>;
+  using B::B;
+};
+
+template <typename T>
+struct ModelCompilationOptionsImpl : ConstModelCompilationOptionsImpl<T> {
+  using B = ConstModelCompilationOptionsImpl<T>;
+  using B::B;
+
+  ModelCompilationOptionsImpl& SetInputModelPath(const ORTCHAR_T* input_model_path);  ///< Wraps OrtApi::ModelCompilationOptions_SetInputModelPath
+  ModelCompilationOptionsImpl& SetInputModelFromBuffer(const void* input_model_data,
+                                                       size_t input_model_data_size);   ///< Wraps OrtApi::ModelCompilationOptions_SetInputModelFromBuffer
+  ModelCompilationOptionsImpl& SetEpContextEmbedMode(bool embed_ep_context_in_model);   ///< Wraps OrtApi::ModelCompilationOptions_SetEpContextEmbedMode
+  ModelCompilationOptionsImpl& SetOutputModelPath(const ORTCHAR_T* output_model_path);  ///< Wraps OrtApi::ModelCompilationOptions_SetOutputModelPath
+  ModelCompilationOptionsImpl& SetOutputModelExternalInitializersFile(const ORTCHAR_T* file_path,
+                                                                      size_t initializer_size_threshold);  ///< Wraps OrtApi::ModelCompilationOptions_SetOutputModelExternalInitializersFile
+  ModelCompilationOptionsImpl& SetOutputModelBuffer(OrtAllocator* allocator, void** output_model_buffer_ptr,
+                                                    size_t* output_model_buffer_size_ptr);  ///< Wraps OrtApi::ModelCompilationOptions_SetOutputModelBuffer
+};
+}  // namespace detail
+
+using UnownedModelCompilationOptions = detail::ModelCompilationOptionsImpl<detail::Unowned<OrtModelCompilationOptions>>;
+using ConstModelCompilationOptions = detail::ConstModelCompilationOptionsImpl<detail::Unowned<const OrtModelCompilationOptions>>;
+
+/** \brief Wrapper around ::OrtModelCompilationOptions
+ *
+ */
+struct ModelCompilationOptions : detail::ModelCompilationOptionsImpl<OrtModelCompilationOptions> {
+  explicit ModelCompilationOptions(std::nullptr_t) {}                              ///< Create an empty ModelCompilationOptions object, must be assigned a valid one to be used.
+  ModelCompilationOptions(const Env& env, const SessionOptions& session_options);  ///< Wraps OrtApi::CreateModelCompilationOptionsFromSessionOptions
+  ModelCompilationOptions(const Env& env, ConstSessionOptions session_options);    ///< Wraps OrtApi::CreateModelCompilationOptionsFromSessionOptions
+  explicit ModelCompilationOptions(OrtModelCompilationOptions* p) : ModelCompilationOptionsImpl<OrtModelCompilationOptions>{p} {}
+  UnownedModelCompilationOptions GetUnowned() const { return UnownedModelCompilationOptions{this->p_}; }
```

Comment:
Is GetUnowned used anywhere? The existing C++ API seem to have minimal usage of GetUnowned so wondering if we can skip adding it unless required.

Reply:
Removed.